### ISSUE TITLE
Cannot stage file through magit-dispatch-popup

### ISSUE THREAD

**Issue by mrhmouse** [magit/magit#2715]
- Make changes to a file
- `C-x M-g`, which runs `magit-dispatch-popup`
- `s`, to stage the file

I then get this error message:
> Wrong type argument: arrayp, nil

Running `M-x magit-stage-file` does work.
Magit version is `20160704.1629`
Emacs version is `24.4.1`

**Comment by npostavs** [comment]
When not compiled the error message is more informative, basic problem is that `magit-stage` assumes it's in a status buffer, and tries to look for current section.
```
Debugger entered--Lisp error: (error "magit-section-type accessing a non-magit-section")
  signal(error ("magit-section-type accessing a non-magit-section"))
  error("%s accessing a non-%s" magit-section-type magit-section)
  (or (and (vectorp section) (>= (length section) 14) (memq (aref section 0) cl-struct-magit-section-tags)) (error "%s accessing a non-%s" (quote magit-section-type) (quote magit-section)))
  (progn (or (and (vectorp section) (>= (length section) 14) (memq (aref section 0) cl-struct-magit-section-tags)) (error "%s accessing a non-%s" (quote magit-section-type) (quote magit-section))) (aref section 1))
  (let* ((val (progn (or (and (vectorp section) (>= (length section) 14) (memq (aref section 0) cl-struct-magit-section-tags)) (error "%s accessing a non-%s" (quote magit-section-type) (quote magit-section))) (aref section 1)))) (cond ((memq val (quote (hunk file))) (let nil section)) ((memq val (quote (staged unstaged untracked stashed-index stashed-worktree stashed-untracked))) (let nil (progn (or (and (vectorp section) (>= ... 14) (memq ... cl-struct-magit-section-tags)) (error "%s accessing a non-%s" (quote magit-section-children) (quote magit-section))) (aref section 13)))) (t (let nil (user-error "Cannot apply this, it's not a change")))))
  (let ((section (magit-current-section))) (let* ((val (progn (or (and (vectorp section) (>= ... 14) (memq ... cl-struct-magit-section-tags)) (error "%s accessing a non-%s" (quote magit-section-type) (quote magit-section))) (aref section 1)))) (cond ((memq val (quote (hunk file))) (let nil section)) ((memq val (quote (staged unstaged untracked stashed-index stashed-worktree stashed-untracked))) (let nil (progn (or (and ... ... ...) (error "%s accessing a non-%s" ... ...)) (aref section 13)))) (t (let nil (user-error "Cannot apply this, it's not a change"))))))
  (or (magit-region-sections (quote hunk) (quote file)) (let ((section (magit-current-section))) (let* ((val (progn (or (and ... ... ...) (error "%s accessing a non-%s" ... ...)) (aref section 1)))) (cond ((memq val (quote (hunk file))) (let nil section)) ((memq val (quote (staged unstaged untracked stashed-index stashed-worktree stashed-untracked))) (let nil (progn (or ... ...) (aref section 13)))) (t (let nil (user-error "Cannot apply this, it's not a change")))))))
  magit-apply--get-selection()
  (let ((it (magit-apply--get-selection))) (if it (progn (let* ((val (list (magit-diff-type) (magit-diff-scope)))) (if (consp val) (let* ((xcar ...) (xcdr ...)) (cond (... ...) (... ...) (... ...) (... ...) (... nil) (... ...) (t nil))) nil)))))
  magit-stage(nil)
  call-interactively(magit-stage)
  (let ((magit-current-popup magit-this-popup) (magit-current-popup-args (magit-popup-get-args)) (command (progn (or (and (vectorp (or action variable)) (>= (length ...) 7) (memq (aref ... 0) cl-struct-magit-popup-event-tags)) (error "%s accessing a non-%s" (quote magit-popup-event-fun) (quote magit-popup-event))) (aref (or action variable) 4)))) (if action (progn (magit-popup-quit))) (call-interactively command) (setq this-command command) (if action nil (magit-refresh-popup-buffer)))
  (if (or action variable) (let ((magit-current-popup magit-this-popup) (magit-current-popup-args (magit-popup-get-args)) (command (progn (or (and (vectorp ...) (>= ... 7) (memq ... cl-struct-magit-popup-event-tags)) (error "%s accessing a non-%s" (quote magit-popup-event-fun) (quote magit-popup-event))) (aref (or action variable) 4)))) (if action (progn (magit-popup-quit))) (call-interactively command) (setq this-command command) (if action nil (magit-refresh-popup-buffer))) (if (eq event 113) (progn (magit-popup-quit) (if magit-previous-popup (progn (magit-popup-mode-setup magit-previous-popup nil)))) (user-error "%c isn't bound to any action" event)))
  (let ((action (magit-popup-lookup event :actions)) (variable (magit-popup-lookup event :variables))) (if (and variable (not (progn (or (and (vectorp variable) (>= ... 7) (memq ... cl-struct-magit-popup-event-tags)) (error "%s accessing a non-%s" (quote magit-popup-event-arg) (quote magit-popup-event))) (aref variable 3)))) (progn (setq action variable) (setq variable nil))) (if (or action variable) (let ((magit-current-popup magit-this-popup) (magit-current-popup-args (magit-popup-get-args)) (command (progn (or (and ... ... ...) (error "%s accessing a non-%s" ... ...)) (aref (or action variable) 4)))) (if action (progn (magit-popup-quit))) (call-interactively command) (setq this-command command) (if action nil (magit-refresh-popup-buffer))) (if (eq event 113) (progn (magit-popup-quit) (if magit-previous-popup (progn (magit-popup-mode-setup magit-previous-popup nil)))) (user-error "%c isn't bound to any action" event))))
  magit-invoke-popup-action(115)
  call-interactively(magit-invoke-popup-action nil nil)
  command-execute(magit-invoke-popup-action)
```

**Comment by npostavs** [comment]
Should be fixed by #2716.
